src/main/webapp/script.js

```diff
@@ -13,7 +13,58 @@
 // limitations under the License.
 
 function addThreads() {  // eslint-disable-line
-  fetch('/data').then((response) => response.text()).then((quote) => {
-    document.getElementById('thread-container').innerText = quote;
+  const url = '/data';
+  fetch(url).then((response) => response.json()).then((threadInfoList) => {
+    const threadList = document.getElementById('thread-container');
+    threadList.innerHTML = '';
+
+    let i;
+    for (i = 0; i < threadInfoList.length; i += 4) {
+      // create description
+      const ulid = 'ul' + i;
+      const lidescription = document.createElement('LI');
+      lidescription.innerText = 'Description';
+      lidescription.className = 'descriptionli';
+      const description = document.createElement('UL');
+      description.appendChild(lidescription);
+      description.appendChild(
+          createListElement('sentiment:' + threadInfoList[i + 1]));
+      description.appendChild(
+          createListElement('upvotes: ' + threadInfoList[i + 2]));
+      description.appendChild(linkListElement(threadInfoList[i + 3]));
+      description.className = ('description');
+      description.setAttribute('id', ulid);
+      // create description
```

Comment:
nit: create button?

Reply:
I am trying to create a method that lets the user toggle between opening and closing the description by clicking the button.